server/src/main/java/io/crate/role/Roles.java

```diff
@@ -61,6 +61,24 @@ default Role findUser(String userName) {
         return null;
     }
 
+    /**
+     * Finds a user by JWT token payload (issuer and username).
+     */
+    @Nullable
+    default Role findUser(String issuer, String username) {
```

Comment:
Could we maybe have a `findUser(Predicate<Role> predicate)` variant instead, and have the jwt specific filter created from elsewhere?

As it is, it looks like all of the following components have become JWT aware:

- Credentials
- HttpAuthUpstreamHandler
- SqlHttpHandler
- HostBasedAuthentication
- Roles
- Role

It would be great if we could isolate that some more

Reply:
https://github.com/crate/crate/pull/15551/commits/741e92bbb8753a254f80be67ac85c33eae9ef1a7

Removed jwt related stuff from `Roles`, `HttpAuthUpstreamHandler` and `SqlHttpHandler`

Still jwt aware:
`Role` - because it has metadata
`Credentials` - it's entry point and used in `AuthenticationMethod`
`HostBasedAuthentication` - we need concrete instance of jwt auth method in methodForName